tutorial/part8.md

```diff
@@ -0,0 +1,792 @@
+
+First add a new component to `./src/state/components.js` called `IsPickup`.
+
+```diff
+export class IsOpaque extends Component {}
+
++export class IsPickup extends Component {}
+
+export class IsRevealed extends Component {}
+```
+
+This component will actually do some things later but for now it's just a flag to know that an entity with it can be picked up.
+
+Register it in `./src/state/ecs.js`:
+
+```diff
+import {
+  ...
+  IsOpaque
++  IsPickup,
+  IsRevealed
+  ...
+} from "./components";
+```
+
+```diff
+ecs.registerComponent(IsOpaque);
++ecs.registerComponent(IsPickup);
+ecs.registerComponent(IsRevealed);
+```
+
+Next we'll create a couple of new prefabs - the generic base `Item` and our more specific `HealthPotion` in `./src/state/prefabs.js`.
+
+```javascript
+export const Item = {
+  name: "Item",
+  components: [
+    { type: "Appearance" },
+    { type: "Description" },
+    { type: "Layer300" },
+    { type: "IsPickup" },
+  ],
+};
+
+export const HealthPotion = {
+  name: "HealthPotion",
+  inherit: ["Item"],
+  components: [
+    {
+      type: "Appearance",
+      properties: { char: "!", color: "#DAA520" },
+    },
+    {
+      type: "Description",
+      properties: { name: "health potion" },
+    },
+  ],
+};
+```
+
+Notice we place our Item prefab on layer300. This is our 'items' layer and will ensure they render above the floor but below players and monsters.
+
+Register these as well, taking care to register the base `Item` prefab before the more specific `HealthPotion`. Remember that prefabs must be registered before they can be inherited from.
+
+```diff
+import {
+  Being,
++  Item,
+  Tile,
++  HealthPotion,
+  Goblin,
+  Player,
+  Wall,
+  Floor,
+} from "./prefabs";
+```
+
+```diff
+// register "base" prefabs first!
+ecs.registerPrefab(Being);
++ecs.registerPrefab(Item);
+
++ecs.registerPrefab(HealthPotion);
+ecs.registerPrefab(Wall);
+```
+
+And finally let's actually add them to the map in `./src/index.js`:
+
+```diff
+times(5, () => {
+  const tile = sample(openTiles);
+  ecs.createPrefab("Goblin").add(Position, { x: tile.x, y: tile.y });
+});
+
++times(5, () => {
++  const tile = sample(openTiles);
++  ecs.createPrefab("HealthPotion").add(Position, { x: tile.x, y: tile.y });
++});
+```
+
+I found it useful during this part to render a ton of potions - maybe 50 - and no goblins. This makes testing a lot easier as you don't have to explore the dungeon to find a potion or worry about getting killed on the way.
+
+Run the game! You should see some potions lying on the dungeon floor (!)
+
+## Adding an inventory
+
+Of course our @ needs a place to store the things it picks up. Time to add an inventory!
+
+Yet again, we'll start by adding a new component. In `./src/state/components.js` add an `Inventory`. It will only need a list property for storing our items.
+
+```javascript
+export class Inventory extends Component {
+  static properties = {
+    list: [],
```

Comment:
geotic allows for an entity array

it would look like:

```js
static properties = {
    list: '<EntityArray>'
}
```

and then you can say

```js
character.inventory.list.add(otherEntity);
```

When it's serialized, it will automatically save just the IDs, and if `otherEntity` is destroyed, it will be removed from the array automatically


i think this feature might be too advanced for this tutorial 🤷

Reply:
This tutorial already feels like it's veering into advanced territory -  I don't think it would be too much. 

I'm trying to think of this as a rough draft with the hope that I'll have enough energy in the end to clean up some of these patterns.

Reply:
As a note - (at least in the current version 3.3.0) - you have to use `character.inventory.list.push` not `character.inventory.list.add`.

`character.inventory.list` is an array. Everything else works great and I'm updating the tutorial to use this.